gmso/core/angle.py

```diff
@@ -31,6 +31,16 @@ class Angle(Connection):
         default=None, description="AngleType of this angle."
     )
 
+    restraint_: Optional[dict] = Field(
```

Comment:
Is the plan to add this field to bonds and impropers as well? I see it in dihedrals.

Reply:
I think we can add this for bond but I am not sure about improper (looking at this https://manual.gromacs.org/current/reference-manual/topologies/topology-file-formats.html and only bond, angle, and dihedral have the restraint option, I am not sure if the dihedral restraint is also applicable to improper)

Reply:
a bond restraint is added. Right now it only works with harmonic potential restraint (though gromacs seems to offer a few more type of distance restraint, we can consider support those in the future)